sdk/src/azure/platform/az_posix.c

```diff
@@ -5,18 +5,53 @@
 #include <azure/core/internal/az_config_internal.h>
 #include <azure/core/internal/az_precondition_internal.h>
 
+#include <errno.h>
+#include <pthread.h>
+#include <signal.h>
+#include <stdlib.h>
 #include <time.h>
-
 #include <unistd.h>
 
 #include <azure/core/_az_cfg.h>
 
+#ifdef __linux__
+#define _az_PLATFORM_POSIX_CLOCK_ID CLOCK_BOOTTIME
+#else
+#define _az_PLATFORM_POSIX_CLOCK_ID CLOCK_MONOTONIC
+#endif
+
+#ifndef __APPLE__
```

Comment:
I think that we should instead have a different file for Apple. The POSIX version that they support can be extracted in a common file and the rest (i.e. timers) can be separated into az_apple maybe, with no implementation right now.